packages/devextreme/testing/tests/DevExpress.ui.widgets/selection.test.js

```diff
@@ -2442,6 +2442,58 @@ QUnit.module('onSelectionChanging', {
         assert.strictEqual(selection.isItemSelected(this.data[0]), true, 'isItemSelected returns true');
     });
 
+    QUnit.test('should be called with correct parameters when select all ', function(assert) {
+        const selectionChangingHandler = sinon.spy();
+        const dataKeys = this.data.map((item) => item.id);
+
+        this.dataSource = createDataSource(this.data, { key: 'id' }, {});
+
+        const selection = new Selection({
+            ...this.basicSelectionConfig,
+            load: () => this.dataSource.load(),
+            onSelectionChanging: selectionChangingHandler,
+        });
+
+        selection.selectAll();
```

Comment:
Please also test `selectedItemKeys` method - it's also part of a Selection module's public API